pystiche/papers/ulyanov_et_al_2016/core.py

```diff
@@ -0,0 +1,205 @@
+from typing import Union, Optional, Callable
+import torch
+from torch.utils.data import DataLoader
+from torch import nn
+from torch.optim.optimizer import Optimizer
+from torch.optim import lr_scheduler
+import pystiche
+from pystiche.optim import (
+    OptimLogger,
+    default_transformer_optim_loop,
+)
+from ..common_utils import batch_up_image
+from .modules import ulyanov_et_al_2016_transformer
+from .loss import (
+    UlyanovEtAl2016PerceptualLoss,
+    ulyanov_et_al_2016_perceptual_loss,
+)
+
+from .data import (
+    ulyanov_et_al_2016_style_transform,
+    ulyanov_et_al_2016_content_transform,
+    ulyanov_et_al_2016_dataset,
+    ulyanov_et_al_2016_image_loader,
+    ulyanov_et_al_2016_images,
+)
+from .utils import (
+    ulyanov_et_al_2016_preprocessor,
+    ulyanov_et_al_2016_postprocessor,
+    ulyanov_et_al_2016_optimizer,
+)
+
+__all__ = [
+    "ulyanov_et_al_2016_transformer",
+    "ulyanov_et_al_2016_perceptual_loss",
+    "ulyanov_et_al_2016_dataset",
+    "ulyanov_et_al_2016_image_loader",
+    "ulyanov_et_al_2016_training",
+    "ulyanov_et_al_2016_images",
+    "ulyanov_et_al_2016_stylization",
+]
+
+
+def ulyanov_et_al_2016_transformer_epoch_optim_loop(
```

Comment:
We should implement a `default_transformer_epoch_optim_loop()` and just parametrize it here if necessary. I will take that on and get back to you if I'm finished.